server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceServiceTests.java

```diff
@@ -414,6 +414,7 @@ private static boolean executeAndReroute(
     ) {
         return desiredBalanceService.updateDesiredBalanceAndReroute(
             new DesiredBalanceInput(
+                0,
```

Comment:
Hmm some of these tests execute this more than once, and in prod we would see a newer index each time. I think that means we're missing assertions somewhere to say that the index should be strictly increasing. I don't see an easy place to put them tho. Maybe `DesiredBalance` itself should record the index from which it was computed? I have a feeling this will be useful later on, at least in debugging. Up to you, we can leave it for now too.

Reply:
I guess this could be asserted inside of `org.elasticsearch.cluster.routing.allocation.allocator.ContinuousComputation#onNewInput`

Reply:
Yeah I wondered about that but `ContinuousComputation` doesn't know that the computation object has a notion of "index".